server/routes/wishCards.js

```diff
@@ -63,512 +54,485 @@ const WishCardController = require('./controller/wishcard.controller');
+    const agency = wishcard.belongsTo;
+
+    // fee for processing item. 3% charged by stripe for processing each card trasaction + 5% from us to cover the possible item price change difference
+    const processingFee = 1.08;
+    // we are using amazon prime so all shipping is free
+    const shipping = 'FREE';
+    // Open for discussion. Each state has its own tax so maybe create values for each individual(key-value) or use a defined one for everything since we are
+    // doing all the shopping
+    const tax = 1.0712;
+
+    const totalItemCost = await calculateWishItemTotalPrice(wishcard.wishItemPrice);
+
+    const extendedPaymentInfo = {
+      processingFee: (wishcard.wishItemPrice * processingFee - wishcard.wishItemPrice).toFixed(2),
+      shipping,
+      tax: (wishcard.wishItemPrice * tax - wishcard.wishItemPrice).toFixed(2),
+      totalItemCost,
+      agency,
+    };
+
+    // console.log(agency);
+
+    // I test printed the agency and wishcard object
+    // and if we are still using agency array, the matching obj is the 1st one
+    // but who knows ¯\_(ツ)_/¯
+
+    res.status(200).render('donate', {
+      user: res.locals.user,
+      wishcard: wishcard || [],
+      extendedPaymentInfo,
+
+      agency,
+    });
+  } catch (error) {
+    handleError(res, 400, error);
+  }
+});
 
 // @desc    display 3 wishcards randomly in the sample wishcard section of index.html
 // @route   GET '/wishcards/get/random'
 // @access  Public
 // @tested 	No
 router.get('/get/random', async (req, res) => {
-	try {
-		// let wishcards = await WishCardRepository.getAllWishCards();
-		let wishcards = [];
-		wishcards = await WishCardRepository.getWishCardsByStatus('published');
-		console.log(wishcards.length);
-		if (!wishcards || wishcards.length < 6) {
-			const donatedWishCards = await WishCardRepository.getWishCardsByStatus('donated');
-			wishcards = wishcards.concat(donatedWishCards.slice(0, 6 - wishcards.length));
-		} else {
-			wishcards.sort(() => Math.random() - 0.5);
-			const requiredLength = 3 * Math.ceil(wishcards.length / 3);
-			const rem = requiredLength - wishcards.length;
-			if (rem !== 0 && wishcards.length > 3) {
-				for (let j = 0; j < rem; j++) {
-					wishcards.push(wishcards[j]);
-				}
-			}
-		}
-		res.render('templates/homeSampleCards', { wishcards }, (error, html) => {
-			if (error) {
-				res.status(400).json({ success: false, error });
-			} else {
-				res.status(200).send(html);
-			}
-		});
-	} catch (error) {
-		handleError(res, 400, error);
-	}
+  try {
+    // let wishcards = await WishCardRepository.getAllWishCards();
+    let wishcards = [];
+    wishcards = await WishCardRepository.getWishCardsByStatus('published');
+    console.log(wishcards.length)
+    if (!wishcards || wishcards.length < 6) {
+
+      const donatedWishCards = await WishCardRepository.getWishCardsByStatus('donated');
+      wishcards = wishcards.concat(donatedWishCards.slice(0, 6 - wishcards.length))
```

Comment:
why are you removing ;?

Reply:
old server, that's from merging the master